gammapy/irf/psf/psf_map.py

```diff
@@ -80,138 +75,216 @@ def psf_map(self):
     def psf_map(self, value):
         self._irf_map = value
 
-    def get_energy_dependent_table_psf(self, position=None):
-        """Get energy-dependent PSF at a given position.
-
-        By default the PSF at the center of the map is returned.
+    @classmethod
+    def from_geom(cls, geom):
+        """Create psf map from geom.
 
         Parameters
         ----------
-        position : `~astropy.coordinates.SkyCoord`
-            the target position. Should be a single coordinates
+        geom : `Geom`
+            PSF map geometry.
 
         Returns
         -------
-        psf_table : `~gammapy.irf.EnergyDependentTablePSF`
-            the table PSF
+        psf_map : `PSFMap`
+            Point spread function map.
         """
-        if position is None:
-            position = self.psf_map.geom.center_skydir
+        geom_exposure = geom.squash(axis_name="rad")
+        exposure_psf = Map.from_geom(geom_exposure, unit="m2 s")
+        psf_map = Map.from_geom(geom, unit="sr-1")
+        return cls(psf_map, exposure_psf)
 
-        if position.size != 1:
-            raise ValueError(
-                "EnergyDependentTablePSF can be extracted at one single position only."
-            )
+    def to_region_nd_map(self, region):
+        """Convert to region ND PSF map
```

Comment:
Maybe  "Returns a PSFMap extracted in a region."

Maybe explain that this should work only with a `WcsGeom` (or `HpxGeom`?)